### PR TITLE
Courses data model with validations

### PR DESCRIPTION
**Pull request by mwitte98**
Related Issue #176 

Changes:
- Courses data model with validations
  - Lesson can't be in a course twice
  - Course instructor can only add the lessons they own
  - Student can't be in a course twice
  - Student can't be in a course they're the instructor for

### CODE REVIEW

File: app/models/course_lesson.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class CourseLesson < ApplicationRecord
```
Comment (djrenren) [inline app/models/course_lesson.rb:1]: Couldn't we just put `belongs_to :course, optional: true, class_name: 'Course'` on `Lesson` and skip this entire join table? seeing as a lesson can only be in one Course.

File: app/models/course_student.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class CourseStudent < ApplicationRecord
```
Comment (djrenren) [inline app/models/course_student.rb:1]: We should probably add a `has_many`...`through` to user so we can access courses that way.

**Review by djrenren — CHANGES_REQUESTED** [review]

File: app/models/course_student.rb (hunk 2 above)
Comment (mwitte98) [reply app/models/course_student.rb:1]: Yeah forgot about that...will do

**Comment by mwitte98** [comment]
Updated to access courses from lesson and user

**Comment by djrenren** [comment]
Please update the seed so we have a sample. otherwise looks good

**Comment by mwitte98** [comment]
@djrenren updated seed

**Comment by djrenren** [comment]
Sorry for being an asshole but can we put a lesson in it too? Maybe just a copy of A's.

**Comment by mwitte98** [comment]
@djrenren no problem. Done.

**Review by djrenren — APPROVED** [review]

**Pull request merged**